pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -221,47 +221,73 @@ func (p *DNSProxy) checkRestored(endpointID uint64, destPort uint16, destIP stri
 // GetRules creates a fresh copy of EP's DNS rules to be stored
 // for later restoration.
 func (p *DNSProxy) GetRules(endpointID uint16) (restore.DNSRules, error) {
+	// Lock Ordering Note: Acquiring the IPCache look (as LookupIPsBySecID does)
```

Comment:
I'm wondering if we'd feel more confident if we had unit tests validating this function, in case this change subtly introduced a regression.